kong/tracing/request_id.lua

```diff
@@ -0,0 +1,44 @@
+local ngx = ngx
+
+local NGX_VAR_PHASES = {
+  set           = true,
+  rewrite       = true,
+  access        = true,
+  content       = true,
+  header_filter = true,
+  body_filter   = true,
+  log           = true,
+  balancer      = true,
+}
+
+
+local function get_ctx_request_id()
+  return ngx.ctx.request_id
+end
+
+
+local function get()
+  local rid = get_ctx_request_id()
+
+  if not rid then
+    local phase = ngx.get_phase()
+    if not NGX_VAR_PHASES[phase] then
```

Comment:
Need to think about if this `if-stmt` checking is necessary, I noticed that you have already use the ngx.var.request_id reference. If the upper user call this function and cannot get ngx.var.request_id, it will get a lua runtime error.

Reply:
not sure I follow, sorry. The only access to ngx.var in this file is protected by this `if` check which verifies whether the current phase can access `ngx.var`, where is the reference you think might be problematic?

Reply:
ok no problem. I know you want to report error safely, which is ok.